src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,59 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+#ifdef ESCARGOT_ENABLE_ES2015
+static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(thisObject, RegExp, match);
+
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+    RegExpObject* rx;
+
+    if (thisObject->isRegExpObject(state) == true) {
```

Comment:
The only restriction by the standard to the `this value` is only to be an object. I think this `toRegExp` conversion is invalid.